core/repos/repo.go

```diff
@@ -0,0 +1,210 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2019 CERN and copyright holders of ALICE O².
+ * Author: Kostas Alexopoulos <kostas.alexopoulos@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package repos
+
+import (
+	"errors"
+	"github.com/spf13/viper"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+	"strings"
+)
+
+type Repo struct {
+	HostingSite string
+	User string
+	RepoName string
+	Revision string
+	Default bool
+}
+
+func NewRepo(repoPath string) (*Repo, error){
+
+	revSlice := strings.Split(repoPath, "@")
+
+	var repoUrlSlice []string
+	var revision string
+	if len(revSlice) == 2 { //revision specified
+		repoUrlSlice = strings.Split(revSlice[0], "/")
+		revision = revSlice[1]
+	} else if len(revSlice) == 1{ //no revision specified
+		repoUrlSlice = strings.Split(revSlice[0], "/")
```

Comment:
I'd advise validating the length of `repoUrlSlice` and returning error if `!= 3`, otherwise the `[]` in the return statement might fail.